internal/storage/bucket_handle.go

```diff
@@ -311,3 +311,47 @@ func (b *bucketHandle) UpdateObject(ctx context.Context, req *gcs.UpdateObjectRe
 
 	return
 }
+
+func (b *bucketHandle) ComposeObjects(ctx context.Context, req *gcs.ComposeObjectsRequest) (o *gcs.Object, err error) {
+	dstObj := b.bucket.Object(req.DstName)
+
+	if req.DstGenerationPrecondition != nil {
+		dstObj = dstObj.If(storage.Conditions{GenerationMatch: *req.DstGenerationPrecondition})
+	}
+	if req.DstMetaGenerationPrecondition != nil {
+		dstObj = dstObj.If(storage.Conditions{MetagenerationMatch: *req.DstMetaGenerationPrecondition})
+	}
+
+	// Converting the req.Sources list to a list of storage.ObjectHandle as expected by the Go Storage Client.
+	var srcObjList []*storage.ObjectHandle
+	for _, src := range req.Sources {
+		currSrcObj := b.bucket.Object(src.Name)
+		// Switching to requested Generation of the object.
+		if src.Generation != 0 {
+			currSrcObj = currSrcObj.Generation(src.Generation)
+		}
+		srcObjList = append(srcObjList, currSrcObj)
+	}
+
+	// Composing Source Objects to Destination Object using Composer created through Go Storage Client.
+	attrs, err := dstObj.ComposerFrom(srcObjList...).Run(ctx)
+	if err != nil {
+		switch ee := err.(type) {
+		case *googleapi.Error:
+			if ee.Code == http.StatusPreconditionFailed {
+				err = &gcs.PreconditionError{Err: ee}
+			}
```

Comment:
can you add more detail in the cl descriptions interms of what error is throw if src.Generation is not matched and dst.generation or dst.metageneration doesn't match.

Reply:
fixed

Reply:
If generation will not match then it is throwing Notfound error, for MetagenerationPrecondition it is throwing Precondition error. - can you be more specific here where you are talking about sourceObject generation or destination object generation